AdyenActions/Components/Redirect/RedirectComponent.swift

```diff
@@ -34,36 +45,94 @@ public final class RedirectComponent: ActionComponent, DismissableComponent {
     ///
     /// - Parameter url: The URL through which the application was opened.
     /// - Returns: A boolean value indicating whether the URL was handled by the redirect component.
     @discardableResult
     public static func applicationDidOpen(from url: URL) -> Bool {
-        UniversalRedirectComponent.applicationDidOpen(from: url)
+        RedirectListener.applicationDidOpen(from: url)
     }
     
-    /// :nodoc:
-    internal lazy var universalRedirectComponent: UniversalRedirectComponent = {
-        let component = UniversalRedirectComponent(style: style)
+    // MARK: - Http link handling
+
+    private func openHttpSchemeUrl(_ action: RedirectAction) {
+        // Try to open as a universal app link, if it fails open the in-app browser.
+        appLauncher.openUniversalAppUrl(action.url) { [weak self] success in
+            guard let self = self else { return }
+            self.registerRedirectBounceBackListener(action)
+            if success {
+                self.delegate?.didOpenExternalApplication(self)
+            } else {
+                self.openInAppBrowser(action)
+            }
+        }
+    }
+
+    private func openInAppBrowser(_ action: RedirectAction) {
+        let component = BrowserComponent(url: action.url, style: style)
         component.delegate = self
-        return component
-    }()
+        component._isDropIn = _isDropIn
+        component.environment = environment
+        browserComponent = component
+
+        presentationDelegate?.present(component: component, disableCloseButton: false)
+    }
     
-    /// :nodoc:
-    private let style: RedirectComponentStyle?
+    // MARK: - Custom scheme link handling
+
+    private func openCustomSchemeUrl(_ action: RedirectAction) {
+        appLauncher.openCustomSchemeUrl(action.url) { [weak self] success in
+            guard let self = self else { return }
+            if success {
+                self.registerRedirectBounceBackListener(action)
+                self.delegate?.didOpenExternalApplication(self)
+            } else {
+                self.delegate?.didFail(with: RedirectComponent.Error.appNotFound, from: self)
+            }
+        }
+    }
     
-    /// :nodoc:
-    private let componentName = "redirect"
+    // MARK: - Helpers
 
+    private func registerRedirectBounceBackListener(_ action: RedirectAction) {
+        RedirectListener.registerForURL { [weak self] returnURL in
+            guard let self = self else { return }
+            
+            let additionalDetails = RedirectDetails(returnURL: returnURL)
+            let actionData = ActionComponentData(details: additionalDetails, paymentData: action.paymentData)
+            self.delegate?.didProvide(actionData, from: self)
+        }
+    }
+    
+}
+
+extension RedirectComponent: Cancellable {
+
+    /// :nodoc:
+    public func didCancel() {
+        dismiss(true) {
+            self.delegate?.didFail(with: ComponentError.cancelled, from: self)
+        }
```

Comment:
if `didCancel` is called, that would mean the `BrowserComponent` is already dismissed, right?